core/starter/src/main/java/org/apache/syncope/core/starter/SyncopeCoreApplication.java

```diff
@@ -105,4 +221,31 @@ public SyncopeCoreStart keymasterStart(final DomainHolder domainHolder) {
     public KeymasterStop keymasterStop() {
         return new KeymasterStop(NetworkService.Type.CORE);
     }
+
```

Comment:
Is introducing these two new interfaces / default implementations the only way to go?

Reply:
Yes, and this approach should be applied to all other bean implementations that do not conform or implement an interface. Specially for event-listeners in the absence of proxies. The `SyncopeCoreInfoContributor` does some very strange things today and is combined to also react to a listener and collect data; This violates many other principals , but I did not change the way it works; mainly the event listener config is moved out of the contributor.

Reply:
For example,

```
    @ConditionalOnMissingBean
    @Bean
    public AuthDataAccessor authDataAccessor() {}
```

Will create a CGLib proxy; since the class is a concrete type and there is no interface to use for the proxy object. The general approach is to turn `AuthDataAccessor` into an interface, with a default implementation class, or something to that effect.

Reply:
I understand the problem now: I'd say then we need to handle this whole item apart for next PR (keeping the current changes in this PR, anyway) - we're talking here about "residual" beans, as several ones were converted into the interface + default impl approach in the recent past.

I'd say that such new interfaces / default impls should be treated properly in separate files, too, rather than being embedded in Application / Configuration classes.

Reply:
Yes, I definitely agree. That would be a fairly comprehensive change, likely to impact many modules and I plan to tackle that separately via a series of PRs. That makes it easier to test incrementally as well, and the only *sure* way I know to test is to look at the beans actuator endpoint and make sure no reference to "CGLIB" is found in the output. It's fairly slow, but certain :)